src/node/blockstorage.cpp

```diff
@@ -734,8 +735,9 @@ bool BlockManager::WriteUndoDataForBlock(const CBlockUndo& blockundo, BlockValid
         // the FindBlockPos function
         if (_pos.nFile < m_last_blockfile && static_cast<uint32_t>(block.nHeight) == m_blockfile_info[_pos.nFile].nHeightLast) {
             FlushUndoFile(_pos.nFile, true);
+        } else if (_pos.nFile == m_last_blockfile && static_cast<uint32_t>(block.nHeight) > m_undo_height_in_last_blockfile) {
```

Comment:
> I must either be missing something about this line or it might be unnecessary. If I understand correctly this is updating `m_undo_height_in_last_blockfile` when the next block is connected after the flush happened. However, why not update it right away when the flush happens? I tried that by removing this line it seems like all tests are still passing without it.

It's not surprising that there's no test coverage for this. But to explain: the goal of the code is to flush the undo file when the height of the highest block in the undo file equals the height of the highest block in the corresponding block file. There are two cases where this can happen. The first case is on line 633 above, where the undo file and block file are flushed together because the undo data is not far behind the block data. The second case is on line 737 above when undo data is delayed and undo file gets flushed after the block file. The `m_undo_height_in_last_blockfile` needs to be set here so the `m_blockfile_info[nFile].nHeightLast == m_undo_height_in_last_blockfile` check in the first case line 633 can work.